### PR TITLE
dev: remove unwraps from crates/pallet and crates/primitives

### PR DESCRIPTION
**Pull request by lambda-0x**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Code style update (formatting, renaming)
- Minor refactor

## What is the current behavior?

Resolves: #165 

## What is the new behavior?

- moved tests to `tests/` folder
- changed return value for `calculate_commitments` to remove unnecessary casting to `H256`
- propagate `unwrap`'s as error where required, and add `expect` where its not possible for `unwrap` to panic which appropriate message

## Does this introduce a breaking change?

Yes

### CODE REVIEW

File: crates/pallets/starknet/src/lib.rs
Comment (EvolveArt) [inline crates/pallets/starknet/src/lib.rs:725]: I'd prefer `InvalidTransaction::Custom` here as it's quite specific to our wrapper and the transaction will NOT be valid in the future.

File: crates/pallets/starknet/src/lib.rs
Comment (EvolveArt) [inline crates/pallets/starknet/src/lib.rs:814]: this would be clearer
```suggestion
        let provided_fee_token_addr: StarkFelt = Self::fee_token_address().into();
```

File: crates/pallets/starknet/src/lib.rs
Comment (EvolveArt) [inline crates/pallets/starknet/src/lib.rs:935]: we should definitely yes, please create a `good first issue`

File: crates/pallets/starknet/src/lib.rs
Comment (EvolveArt) [inline crates/pallets/starknet/src/lib.rs:963]: no need for this once #667 is merged

File: crates/pallets/starknet/src/lib.rs
Comment (EvolveArt) [inline crates/pallets/starknet/src/lib.rs:972]: `.expect` can still panic which is what we want to avoid in the runtime at all cost. Please look into `.map_err` instead

File: crates/primitives/starknet/src/crypto/commitment/mod.rs
Comment (EvolveArt) [inline crates/primitives/starknet/src/crypto/commitment/mod.rs:59]: No need to touch this file on this PR because of #667

**Review by EvolveArt — CHANGES_REQUESTED** [review]
Overall please avoid `.expect` in the runtime, doesn't really solve the issue of potential panicking.

File: crates/pallets/starknet/src/lib.rs
Comment (lambda-0x) [reply crates/pallets/starknet/src/lib.rs:725]: i used `Future` because i saw mention of nonce too high https://crates.parity.io/sp_runtime/transaction_validity/enum.InvalidTransaction.html#variant.Future.

for `Custom` what should be the value of u8?

File: crates/pallets/starknet/src/lib.rs
Comment (lambda-0x) [reply crates/pallets/starknet/src/lib.rs:972]: > expect can still panic

agreed but the invariant here is that `len(receipts)` will always be <= `MaxTransaction` because of the way we are creating it.

 so this would never panic and thus is okay to `unwrap` it. 

i changed it to `except` just to make it more descriptive.

File: crates/primitives/starknet/src/crypto/commitment/mod.rs
Comment (lambda-0x) [reply crates/primitives/starknet/src/crypto/commitment/mod.rs:59]: haha nice, you already fixed that.

File: crates/pallets/starknet/src/lib.rs
Comment (lambda-0x) [reply crates/pallets/starknet/src/lib.rs:963]: okay we can wait till that is merged then

**Comment by lambda-0x** [comment]
blocked on #667

File: crates/pallets/starknet/src/lib.rs
Comment (EvolveArt) [reply crates/pallets/starknet/src/lib.rs:725]: Right but `Future` implies that transaction would be valid in the future which is not right here.

You can do this for `Custom` : `pub(crate) const NONCE_DECODE_FAILURE: u8 = 1;`

File: crates/pallets/starknet/src/lib.rs
Comment (EvolveArt) [reply crates/pallets/starknet/src/lib.rs:972]: Alright

File: crates/pallets/starknet/src/lib.rs
Comment (LucasLvy) [reply crates/pallets/starknet/src/lib.rs:935]: no it should not, we're using the sequencer address in each transaction and converting it from Felt252Wrapper to bytes is expensive. If you want to change it you can change it to starkfelt but i don't think it adds any value and it probably implies a refacto on the blockifier side

**Review by LucasLvy — APPROVED** [review]

**Comment by LucasLvy** [comment]
rebase por favor

**Comment by lambda-0x** [comment]
@LucasLvy this is blocked on the other PR, because @EvolveArt has refactors some stuff in there. So i will rebase and update this once its merged

File: crates/pallets/starknet/src/lib.rs
Comment (LucasLvy) [reply crates/pallets/starknet/src/lib.rs]: The starknet-rs `FieldElement` type is internally using the Montgomery representation which is expensive to convert to regular bytes

Comment (lambda-0x) [reply crates/pallets/starknet/src/lib.rs]: feel free to revert the PR then, or lmk if its not possible to revert it directly

Comment (lambda-0x) [reply crates/pallets/starknet/src/lib.rs]: feel free to revert the PR then, or lmk if its not possible to revert it directly i will open a PR

**Pull request merged**
